core/src/main/java/org/apache/struts2/result/ServletRedirectResult.java

```diff
@@ -250,13 +250,24 @@ protected void doExecute(String finalLocation, ActionInvocation invocation) thro
      * @throws IOException in case of IO errors
      */
     protected void sendRedirect(HttpServletResponse response, String finalLocation) throws IOException {
-        if (SC_FOUND == statusCode) {
-            response.sendRedirect(finalLocation);
-        } else {
-            response.setStatus(statusCode);
-            response.setHeader("Location", finalLocation);
-            response.getWriter().write(finalLocation);
-            response.getWriter().close();
+        try {
+            if (SC_FOUND == statusCode) {
+                response.sendRedirect(finalLocation);
+            } else {
+                response.setStatus(statusCode);
+                response.setHeader("Location", finalLocation);
+                try {
+                    response.getWriter().write(finalLocation);
+                } finally {
+                    response.getWriter().close();
+                }
+            }
+        } catch (IOException ioe) {
+            LOG.warn("Unable to redirect to: {}, code: {}! (IOException): {}", finalLocation, statusCode, ioe.toString());
+            throw ioe;  // Re-throw required to preserve existing default behaviour
+        } catch (IllegalStateException ise) {
+            LOG.warn("Unable to redirect to: {}, code: {}! isCommited: {}. (IllegalStateException): {}", finalLocation, statusCode, response.isCommitted(), ise.toString());
```

Comment:
Please do  not use `.toString`, Log4j2 will do it for you :)

Reply:
Don't put `{}` for the exception.

Reply:
Hi Again.

The reasoning for the `toString()` here was pretty much the same as https://github.com/apache/struts/pull/366#discussion_r326865745.

The same general logic holds here, so we could just output the full stacktrace and then users who want or need different behaviour can override the `ServletRedirectResult` if needed.